pkg/jobs/registry.go

```diff
@@ -1812,6 +1815,23 @@ func (r *Registry) CheckPausepoint(name string) error {
 	return nil
 }
 
+func (r *Registry) RelocateLease(
+	ctx context.Context,
+	txn isql.Txn,
+	id jobspb.JobID,
+	destID base.SQLInstanceID,
+	destSession sqlliveness.SessionID,
+) (sentinel error, failure error) {
```

Comment:
🤷 I can't decide if a function that returns two errors is more or less weird than a function that always returns an error.

Reply:
I know, right? I went back and forth on this and decided I liked this best: I want to indicate _failure_ vs success separately since e.g. callers are encouraged to Wrapf() their failures with contextual info or log them or clean up differently. I thought about making the first have a different type that just happened to implement Error() but I eventually decided `error` made it most obvious that you return this thing from Resume() and don't go make your own.